atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/pool/HostLocationSupplierTest.java

```diff
@@ -42,7 +43,8 @@ public void shouldReturnEc2Location() {
         Supplier<Optional<HostLocation>> hostLocationSupplier =
                 new HostLocationSupplier(ec2SnitchSupplier, ec2LocationSupplier, Optional.empty());
 
-        assertThat(hostLocationSupplier.get()).hasValue(ec2LocationSupplier.get());
+        assertThat(hostLocationSupplier.get())
+                .hasValue(ec2LocationSupplier.get().get());
```

Comment:
Does this work?
```suggestion
        assertThat(hostLocationSupplier.get())
                .isEqualTo(ec2LocationSupplier.get());
```

Reply:
Giving it a shot, but i don't think it will work. The two suppliers are not "equal", the values they produce are.

Reply:
I was wrong, we are comparing Optionals here, not Suppliers :). It works.

Reply:
yep, that was my thought! Agreed that comparing suppliers would have been wrong though.